hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/AbstractMultiOutputCompactor.java

```diff
@@ -79,5 +79,6 @@ protected void abortWriter(T writer) throws IOException {
           e);
       }
     }
+    writer = null;
```

Comment:
Do we still need to set writer to null in this method? Not in the reset method?

Reply:
The reset method was introduced to allow resetting the writer from outside of a compactor, when the new files gets committed as storefiles. We also have reset it in every abortWriter, but since these methods have direct access to the variable I would prefer to do it like this instead of calling the reset method.

Reply:
It is just a bit weird that we set writer to null directly in these methods while we do not in some other methods. Please add some comments to explain the reason, that we will only call reset from upper layer when we successfully commit the writers so if we fail to commit the writers, we need to reset it by our own.

Reply:
Please add some comments here. Thanks.